### PR TITLE
Upgrade Storybook to v6

### PR DESCRIPTION
**Pull request by jeremylenz**
Testing instructions are on the associated Foreman PR: https://github.com/theforeman/foreman/pull/7990

Along with the upgrade to v6, this PR incorporates several Storybook improvements and addresses issues I had along the way:

* Switched to the new [`main.js` API](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#to-mainjs-configuration).  This simplifies our configuration quite a bit.
* With the new `main.js` configuration, it appeared that some Webpack loaders were being run twice on the same files, particularly the `@mdx-js/loader`.  This manifested in an error message `Unexpected default export without title` for every story.  The loaders were being run once from `main.js` and another time from our manual Webpack configuration in `webpack.config.rules.js`.
* After addressing the above, Storybook still wouldn't render any stories, instead spewing out copious console errors.  I noticed that in `index.js` we were doing `export * from ` several `@storybook/addon` packages.  Once I removed them, the Storybook finally rendered.
* Since the Storybook addon exports are no longer passed through `@theforeman/stories`, the stories files in Foreman must import them directly from `@storybook/addon-knobs` etc.  I added deprecation errors to inform the users.  Also, I suppressed the `import/no-extraneous-dependencies` eslint rule for packages that are dependencies of `@theforeman/stories`.
* The story sorting implementation we were using in v5.3 involved passing in a `storyWeight` parameter, which stopped working in v6.  Thanks to the Storybook folks for the help on https://github.com/storybookjs/storybook/issues/11010, I was able to update our `storySort` function and no changes will be required in our story files.

### CODE REVIEW

File: packages/stories/src/index.js [hunk 1]
```diff
@@ -1,5 +1,2 @@
-export { storiesOf } from '@storybook/react';
-export * from '@storybook/addon-knobs';
-export * from '@storybook/addon-actions';
-export * from '@storybook/addon-storysource';
```
Comment (jeremylenz) [inline packages/stories/src/index.js:4]: Storybook didn't render when both of these were exported.  Probably not safe to `export * from` multiple npm packages when some of the exports could have the same names.

**Comment by jeremylenz** [comment]
@sharvit @johnpmitsch rebased and ready for review :)

Will squash once acked

File: packages/stories/src/config/services/sorting.js [hunk 2]
```diff
@@ -6,17 +6,31 @@ export const storyWeight = 1000;
...
   if (storyA.kind === storyB.kind) return 0;
 
-  // use parameters.storyWeight
-  if (storyA.parameters.storyWeight !== storyB.parameters.storyWeight) {
-    return storyA.parameters.storyWeight - storyB.parameters.storyWeight;
+  // use storyWeight
+  const storyWeightA = kindParamsA.storyWeight || globalParamsA.storyWeight;
+  const storyWeightB = kindParamsB.storyWeight || globalParamsB.storyWeight;
+  const combined = storyWeightA - storyWeightB;
+  if (combined) {
```
Comment (jeremylenz) [inline packages/stories/src/config/services/sorting.js:24]: will evaluate to false if `0`

File: packages/stories/src/config/webpack/webpack.config.vendor.js [hunk 3]
```diff
@@ -10,8 +10,9 @@ module.exports = ({ config, mode }) => {
   const vendorJsFiles = getFilenamesWithExt(vendorManifest.files, '.js');
   const vendorCssFiles = getFilenamesWithExt(vendorManifest.files, '.css');
 
-  // htmlWebpackPlugin is the first plugin in the array
-  const htmlWebpackPlugin = config.plugins[0];
+  const htmlWebpackPlugin = config.plugins.find(
+    p => p.constructor.name === 'HtmlWebpackPlugin'
```
Comment (jeremylenz) [inline packages/stories/src/config/webpack/webpack.config.vendor.js:14]: After the upgrade it appears the order of `config.plugins` changed, so I changed this to look for the correct name instead

**Comment by johnpmitsch** [comment]
The functionality appears to be working great! 

The code seems fine but I need to take a closer look at the webpack aspect of it as I'm not as familiar with the existing code there. Maybe @sharvit can review that part better.

**Comment by jeremylenz** [comment]
@sharvit can this be merged soon?

**Comment by johnpmitsch** [comment]
@jeremylenz commits look a little funky here, did something happen when rebasing?

**Comment by jeremylenz** [comment]
@johnpmitsch Re-rebased, hopefully it's better now

**Comment by johnpmitsch** [comment]
@jeremylenz still working well :+1:

**Review by johnpmitsch — APPROVED** [review]

**Review by sharvit — CHANGES_REQUESTED** [review]
Thanks @jeremylenz 👍 

I deployed those changes to `npm` as `6.1.0-next-storybook-6-upgrade.0` and it works well together with https://github.com/theforeman/foreman/pull/7990, see:
- https://github.com/theforeman/foreman-js/releases/tag/v6.1.0-next-storybook-6-upgrade.0

To install it, replace the version in the `package.json` with `^6.1.0-next-storybook-6-upgrade.0`.

1. Please update the readme file according to the changes and the deprecations.

2. Please squash and finalize your commit message.
In your commit, describe the breaking changes and write a migration guide so it will be landed in the changelog (https://github.com/theforeman/foreman-js/blob/master/CHANGELOG.md) and in the release notes (https://github.com/theforeman/foreman-js/releases).

File: packages/stories/src/config/decorators/centered.js [hunk 4]
```diff
@@ -1,15 +0,0 @@
-import addonCentered from '@storybook/addon-centered/react';
-
-// Centered addon causes layout issues on docs page.
-// See: https://github.com/storybookjs/storybook/issues/8128
-export const centered = (...args) => {
```
[the remark this answers is not in the document]
Comment (jeremylenz) [reply packages/stories/src/config/decorators/centered.js:5]: Correct, see https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#removed-addon-centered

**Comment by jeremylenz** [comment]
Thanks @sharvit, squashed and updated!

**Comment by sharvit** [comment]
Thanks @jeremylenz 

1. Please add another depreciation error and a breaking change line about the `storiesOf` is not supported anymore.

2. I was playing with the changelog and the release notes generator and it seems to be working best when I formated the commit as well:
```
    feat(stories): update Storybook to v6

    BREAKING CHANGE: @storybook/addon-knobs, @storybook/addon-actions,
    and @storybook/addon-storysource are no longer re-exported
    from @theforeman/stories.  Going forward, import these items
    directly: e.g. `import { withKnobs } from @storybook/addon-knobs`
    instead of `import { withKnobs } from @theforeman/stories`.
    Deprecation error messages have been added where appropriate.

    BREAKING CHANGE: The Preview component has been renamed to Canvas.

    BREAKING CHANGE: The Props component has been renamed to ArgsTable.

    BREAKING CHANGE: Custom separators are no longer supported in story titles.
    Instead of `title: 'Components|DiffView'` you must write `title: 'Components/DiffView'`.
```

This format generated the following release notes:
- https://github.com/theforeman/foreman-js/releases/tag/v7.0.0-next-storybook-v6.0
- https://github.com/theforeman/foreman-js/blob/v7.0.0-next-storybook-v6.0/CHANGELOG.md#700-next-storybook-v60-2020-11-17

Please format the commit accordingly.

**Comment by jeremylenz** [comment]
@sharvit updated and formatted the commit message!

**Review by sharvit — APPROVED** [review]
Thanks @jeremylenz

**Comment by sharvit** [comment]
Published as v7.0.0, see https://github.com/theforeman/foreman-js/releases/tag/v7.0.0

**Pull request merged**